python/lsst/afw/image/_visitInfo.cc

```diff
@@ -73,7 +74,8 @@ void declareVisitInfo(lsst::utils::python::WrapperCollection &wrappers) {
                         "rotType"_a = RotType::UNKNOWN,
                         "observatory"_a = coord::Observatory(nanAngle, nanAngle, nan),
                         "weather"_a = coord::Weather(nan, nan, nan), "instrumentLabel"_a = "", "id"_a = 0,
-                        "focusZ"_a = nan, "observationType"_a = "");
+                        "focusZ"_a = nan, "observationType"_a = "", "scienceProgram"_a = "",
+                        "observationReason"_a = "", "object"_a = "", "hasSimulatedContent"_a = false);
```

Comment:
Yeah, I worried about that too, but the infrastructure to make this behave as a `None`-able field seemed excessive (I don't think we want have `nullptr`s in VisitInfo). I'll add a note here, and I think to the couple of places in `VisitInfo.cc` that also default to `false`. @timj said that going forward it should always be properly populated for LSST data (and we'll fix the metadata where it's not):

https://lsstc.slack.com/archives/C2JPMCF5X/p1661906553417399?thread_ts=1661881492.710089&cid=C2JPMCF5X

I'm open to suggestions though, since I'm also worried about this causing someone to be misled when using data from other sources.

Reply:
Yes, `ObservationInfo` will always set it so if your VisitInfo is set from raw data that have been processed in gen3 it will always be set and ObservationInfo declares it to be bool -- if someone creates their ObservationInfo and requests that the simulated content flag not be set and then passes it to VisitInfo then it will have a None value. If you are worried that this might be common then I don't know how easy it is to have a True/False/Undef state in C++.

Reply:
A note is all I was asking for; I definitely don't think using something like `std::optional` is worth it!

(Though, given the follow-up discussion, I trust there's a better argument for its correctness than "This constructor must only be called by `ObservationInfo`"? I assumed it was more that any code that didn't set `hasSimulatedContent` was written before there was a need to distinguish simulated from real data.)

Reply:
I don't know if we actually have a use case for `hasSimulatedContent` in the pipeline yet: I suspect it's mostly for notebook analysis of various test stand data that was made to look like it's on-mountain. I've added notes, though I wish I had something more to say than "// default false for backwards compatibility"?

Reply:
Oh, and this reminds me: it currently prints `0` or `1`, since those are the C++ true/false output values. I don't think we want to use `std::boolalpha`, since printing will happen from python and the capitalization will make them unhelpful. So, I'm left with just having them printed as 0 or 1 and it's up to the user to interpret?

Reply:
I'd prefer `boolalpha`, it's more reader-friendly. I think it's easier to interpret "`true` really means `True`" than "`1` really means `True`".